### ISSUE TITLE
websocket doesn't work/connect.

### ISSUE THREAD

**Issue by wiegleyj** [apollo-ng/picoReflow#6]
When I first connect to the server I get two messages:
"Yay, I'm alive" which seems fine and another which says "Error 1: status websocket not available" which is all red and ominous.

I can't see either of the default profiles contained in the storage directory and on the host's terminal I am getting:

2014-12-16 21:20:29,160 ERROR ovenWatcher: could not write to socket <geventwebsocket.websocket.WebSocket object at 0xb4daf110>

I'm not getting sensor readings either. I have a raspberry pi B+ and I changed the config.py to reflect having my adafruit MAX31855 hooked up to the SPI port pins.

**Comment by ScottW1** [comment]
I also had my SSRs turning on as soon as the python program was run. I needed to swap the controls to on=HIGH an off==LOW. The code has on=LOW and off=HIGH as written.  That doesn't help your websocket issue though.

**Comment by ScottW1** [comment]
After running fine for me many times, I am now having the websocket stall on (or after) loading the picoreflow.css (it is the last entry in the log).  Pulling up the webpage, everything loads but the profiles.  So I can not run the system. Tried it on multiple computers/browsers, and they are all the same. Multiple restarts of the Pi to no avail.

Last night I ran a 6 hour run.  The websocket stopped updating the website after ~4 hours, as expected from previous runs. The program ran to completion on the python side, but I was never able to get the webpage to completely load again after that.

Any ideas?
I'll try doing a clean install and see if that makes a difference.

**Comment by chron0** [comment]
Yes, the output mapping needs to be defined by the user as of now, since some systems need a high to have output state = low, it all depends on the hardware connected to it and as such is currently up to the person implementing it to figure out correctly. 

It's most likely the contents stored in the leveldb, so you could basically just checkout the contents of of the the leveldb storage folder again to get a clean slate as opposed to doing everything from scratch. However, have you had a look into the javascript console in the browser to see if there are any obvious error messages indicating what specifically causes the bug?

**Comment by ScottW1** [comment]
I didn't notice any errors. There was a warning about not using a "generic" version of a specific command. But it wasn't an error. It just seemed to be stalled at that point. It kept taking new websocket requests, but then stalled again (not always at the same point though).

**Comment by ScottW1** [comment]
ACH!!!  I had a stupid syntax error from when I was messing with the time unit display.  Its running again! YAY!

The warning I talked about before is:
'webkitRequestAnimationFrame' is vendor-specific. Please use the standard 'requestAnimationFrame' instead.

From:
jquery.flot.resize.js:23 

Could that be causing the issue?

**Comment by chron0** [comment]
Yeah, that has been there all the time, comes from upstream flot. Both are just warnings/deprecation notices, they're not nice but usually didn't mess with operation. But they were also reasons why I wanted to move away from jquery and flot as well :)

**Comment by wiegleyj** [comment]
Sorry for the absence... I downloaded the latest version and things are a bit better but not perfect now.

Better: The max31855 thermocouple is now reading temperature correctly when connected to the SPI pins (I originally designed my electronics to communicate using the natural SPI pins and I used the PWM capable GPIO pin to drive the heating element, I have changed the config.py file to reflect this) So now the temperatures read.

Better: The websocket seems to be working fine now.

Not working still: The oven's on/off state still seems to be reversed. The oven is on when it should be off and vice versa. I read ScottW1's post about "off=HIGH" but I cannot find any such line/position to change this string anywhere in any of the picoReflow files.  And I'm skeptical about this existing. But I did find the self_heat method in oven.py and it seems it wants to turn the oven ON when value is true. and the argument sent to GPIO.Output in the true condition is GPIO.LOW which is the opposite of what I need since my SSR is active high. So I swapped the GPIO.HIGH/GPIO.LOW values.

So I think things are working now. Yay!!!

A suggestion... My unit does not have a door switch or fan.  I configured those pins in the config.py to unconnected GPIO pins in the raspberry pi but I think it would be nice if the software made these two switches and their features entirely optional.

**Comment by ScottW1** [comment]
Yes, you are correct.  I referred to the GPIO pins being pulled high or low.  In the code the syntax is TRUE or FALSE. Sorry for the confusion.

**Comment by wiegleyj** [comment]
I spoke too soon. The "Error 1: Status websocket not available" problem is back but now I have more information.  The raspberry pi/picoReflow is connected to my home network via a wifi usb dongle. I have two workstations that use chrome on to attempt to connect to the picoReflow system. One is a wired desktop and one is a wifi laptop.

The wired desktop works fine and the picoReflow works as expected.

The wifi laptop does not work and throws the "Status websocket not available".

Does anyone know why a wifi-wifi attempt would have this probem and a wired-wifi would not?? very strange.

**Comment by chron0** [comment]
Hmm that doesn't sound like a problem in the software itself, more like a problem with the setup. can you do a netstat -patune on the pi and see if the server is listening on 0.0.0.0 or at least the IP of the wifi interface? In our test setup the Pi was also connected via WiFI (as a client to another AP) and we could use both WiFi and Ethernet clients simultaneously without a problem. I'll try to find which WiFi stick we used, as there are some which were causing some trouble with the PI IIRC.
